### PR TITLE
Add initial settings message to Windows embedding

### PR DESCRIPTION
**Pull request by stuartmorgan**
## Description

Sends the flutter/settings update message to the engine after starting
it. For now values other than 24-hour time preference are hard-coded,
but dark mode support can be added later.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/65590

## Tests

I added the following tests: A new test of engine initialization, including validating that this new message is sent.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [x] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation.
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Reviewer Checklist

- [ ] I have submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[contributor guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[tree hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

**Comment by stuartmorgan** [comment]
@cbracken Ping

**Comment by stuartmorgan** [comment]
At least for now I've turned off `gtest_repeat` (added in https://github.com/flutter/engine/pull/12275) for the desktop embedding unit tests, because the lambda approach we're using is incompatible with multiple runs (I just hadn't noticed yet since we are just starting these tests). If we want to be able to support `gtest_repeat`, we'll need to completely revisit how we handle proctable replacement.

File: shell/platform/windows/flutter_windows_engine.cc [hunk 1]
```diff
@@ -114,10 +118,18 @@ FlutterWindowsEngine::FlutterWindowsEngine(const FlutterProjectBundle& project)
...
   plugin_registrar_->engine = this;
 
+  messenger_wrapper_ = std::make_unique<BinaryMessengerImpl>(messenger_.get());
   message_dispatcher_ =
       std::make_unique<IncomingMessageDispatcher>(messenger_.get());
   window_proc_delegate_manager_ =
       std::make_unique<Win32WindowProcDelegateManager>();
+
+  // Set up internal channels. In the longer term, these should be replaced with
+  // formal embedder APIs.
```
Comment (cbracken) [inline shell/platform/windows/flutter_windows_engine.cc:128]: Not sure if there's an existing bug for this, but if so, consider adding a link to it here and marking this as TODO. If not, we should probably file one; or maybe one for each channel.

**Review by cbracken — APPROVED** [review]
lgtm

**Comment by chinmaygarde** [comment]
I've filed a presubmit flake report and rerun the presubmit.

File: shell/platform/windows/flutter_windows_engine.cc (hunk 1 above)
Comment (stuartmorgan) [reply shell/platform/windows/flutter_windows_engine.cc]: Done

**Pull request merged**